src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -676,34 +679,42 @@ public void setOnlineHandler(BiConsumer<Object, Object[]> onOnline) {
 							this.address.getPath(), this.address.getQuery(), HybridConnectionConstants.Actions.LISTEN,
 							trackingId);
 
-				return delayTask.thenCompose((res) -> {
-					synchronized (this.thisLock) {
-						return this.webSocket.connectAsync(websocketUri, timeout, config);
-					}
-				}).thenRun(() -> this.onOnline());
+				ClientWebSocket webSocket = new ClientWebSocket(this.listener.trackingContext, EXECUTOR);
+				return delayTask.thenCompose(($void) -> {
+					return webSocket.connectAsync(websocketUri, timeout, config).thenApply(($void2) -> webSocket)
+						.whenComplete((ws, err) -> {
+							if (err == null) {
+								this.onOnline();
+							}
```

Comment:
>} [](start = 7, length = 1)

should I put an explicit throw (with trace) in here?
```
.whenComplete((ws, err) -> {
       if (err == null) {
           this.onOnline();
!>   } else {
!>       throw RelayLogger.throwingException(err);
!>   }
```

Reply:
This changed to a thenApply so this comment is out of date.

---
In reply to: [258635516](https://github.com/Azure/azure-relay-java/pull/17#discussion_r258635516) [](ancestors = 258635516)